### PR TITLE
Support optional arguments on Integrated-devnet

### PR DESCRIPTION
**Pull request by Nathan-SL**
- Support optional arguments on integrated-devnet

## Usage related changes
- Optional arguments can now be passed in `integratedDevnet`. 
- Example: `--lite-mode`, `--lite-mode-block-hash` ...
```javascript
integratedDevnet: {
    url: "http://127.0.0.1:5050",
    venv: "active",
    args: ["--lite-mode"]
}
```
## Development related changes
- 

## Checklist:

- [x] I have formatted the code
- [x] I have performed a self-review of the code
- [x] I have rebased to the base branch
- [x] I have documented the changes
- [x] I have updated the tests
- [x] I have created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example).

### CODE REVIEW

File: src/devnet/docker-devnet.ts [hunk 1]
```diff
@@ -38,7 +39,8 @@ export class DockerDevnet extends IntegratedDevnet {
             "-p",
             `${this.host}:${this.port}:${DEVNET_DOCKER_INTERNAL_PORT}`,
             `${this.image.repository}:${this.image.tag}`
-        ]);
+        ].concat(this.args || []);
+        return  spawn("docker", args);
```
Comment (FabijanC) [inline src/devnet/docker-devnet.ts:43]: Rerun prettify

**Review by FabijanC — CHANGES_REQUESTED** [review]
Document this in README.md in the integrated-devnet section (it was expected to be so if the documentation check in the PR description says so). You can just update the hardhat.config sample there with a line and a comment; you can use the same args that you used in the test.

File: src/devnet/docker-devnet.ts (hunk 1 above)
Comment (Nathan-SL) [reply src/devnet/docker-devnet.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/122/commits/b45e5111813159db53cc8d84f1dca17ef44d4851

**Review by FabijanC — APPROVED** [review]
Approving; the tests are failing but should be fixed with 0.9.0 support (blocked by #124).

**Comment by FabijanC** [comment]
Blocked by #124

**Pull request merged**